### ISSUE TITLE
[Topic] Permission checks for certain commands (server owners and server admins).

### ISSUE THREAD

**Issue by GostOmega** [Cog-Creators/Red-DiscordBot#429]
I would like to know if changing [audio.py Line 1113](https://github.com/Twentysix26/Red-DiscordBot/blob/develop/cogs/audio.py#L1113) from `@checks.is_owner()` to `@checks.serverowner_or_permissions(administrator=True)`

Reason I didn't do a pull request was because I rather ask for other opinions if this should changed, likewise someone who is already working on the cog can implement this small change, this is way too small to do a pull request on this one line. If there's a section for requests or ideas on here, link me there and I'll copy and paste this over there, and close this topic.

I'd like this to allow any admins on a server to have the ability to disconnect the bot off a voice channel. In my reasoning, there are more Admins that are on actively throughout the day to monitor the channels, compared to only one Owner who may or may not be busy with other activities outside of their server. Purpose behind this is to prevent the probability of the bot ghosting on the a voice channel, and crashing upon a song request later (this used to happen many months ago, so I'm not sure if this is still a thing) or needing to be kicked and reinvited. 

[Personal Request] To be very specific in my case, I'm running the bot on my girlfriend's server, so I'm not the owner... and she doesn't know (nor wants to learn) any of the commands. It'll be the simple fact that I would do it myself from now on lol.

**Comment by tekulvw** [comment]
The `disconnect` command was originally intended only as a debug tool when writing the audio plugin, though everyone seems to be finding it. The bot will automatically disconnect from a voice channel 5 minutes after all audio in that channel has stopped playing. Is there a real need for a separate `disconnect` command?

**Comment by GostOmega** [comment]
I'm not quite sure what you mean by "separate command", I was only mentioning the permission checks allowing server admins to use that command. However I'll admit that I didn't know it's purpose was for debugging when the bot wouldn't disconnect from voice. If the auto disconnect is working as intended, then I agree with you, there's no point in the use of this command, it's just there as a tool if needed. Thank you for clarifying.

However the conversation I wanted to spark was if admins are seen as second owners of a server (if the Role in that server has Administration enabled). If this is the case, the only commands I believe the admins can benefit from are the following command or groups of commands: audioset, cog, debug, load/reload/unload modules, and shutdown. This is just my opinion, I'd like to know what other people think. If there's any plans on implementing new tools or commands for the bot, I hope this conversation will be put into consideration on permission checks.

**Comment by Kowlin** [comment]
Allow me to add on this by saying that `cog`, `debug` and `load/reload/unload` are all critical commands that _can_ impact systems. They load and execute code on the whole bot.

**Comment by GostOmega** [comment]
Ya in this case I can see the trust issues, especially if your bot is in multiple servers. At first I thought maybe by executing cog or load/reload/unload commands could PM the Owner of the bot for confirmation with the yes/no response, but then again it's no different than the Owner executing the commands themselves, so giving Admins permissions on those wouldn't benefit. I changed my opinion on those commands, and agree not changing debug as well.

I got a question: say a host was running startRedLoop.bat and the bot was running several days without crashing. If the bot was slow at responding to commands and the host was gone all day to work for example, would running the `[p]shutdown` be a useful method?

Also, what's your opinion on Admins having permissions for audioset? Does it effect across all the servers or just in the one server? @Kowlin
